examples/callback-results/src/lib.rs

```diff
@@ -65,3 +65,63 @@ impl Callback {
         (b.is_err(), c.is_err())
     }
 }
+
+#[cfg(all(test, not(target_arch = "wasm32")))]
+mod tests {
+    use tokio::fs;
+    use workspaces::prelude::*;
+
+    #[tokio::test]
+    async fn workspaces_test() -> anyhow::Result<()> {
+        let wasm = fs::read("res/callback_results.wasm").await?;
+
+        let worker = workspaces::sandbox();
+
+        let contract = worker.dev_deploy(wasm).await?;
+
+        // Call function a only to ensure it has correct behaviour
+        let res = contract.call(&worker, "a").transact().await?;
+        assert_eq!(res.json::<u8>()?, 8);
+
+        // Following tests the function call where the `call_all` function always succeeds and handles
+        // the result of the async calls made from within the function with callbacks.
+
+        // No failures
+        let res = contract
+            .call(&worker, "call_all")
+            .args_json((false, 1u8))?
+            .gas(300_000_000_000_000)
```

Comment:
Do we not have a constant for the max prepaid gas somewhere?

Reply:
not currently, maybe a good decision for some cases, alternative may be to have another function called `max_gas` or something to avoid having a constant that they have to find and match themselves.

@ChaoticTempest you might have an idea if any of this would be worth it for workspaces :)

Reply:
the only negative with defaulting max gas is that this API is agnostic of network, so this might make sense with sandbox or testnet, but could lead to issues if the developer is unaware of this default on mainnet.

Either case, this will affect when we update workspaces, so I don't think this is blocking this PR?

Reply:
Def not blocking!

>the only negative with defaulting max gas is that this API is agnostic of network, so this might make sense with sandbox or testnet, but could lead to issues if the developer is unaware of this default on mainnet.

Do we actually inted workspaces to be used with `mainnet`? My gut feeling is that having a single library to be used both as a testing DSL and as the primary way to do stuff in prod is not great, as the incentives for the two use-cases are rather opposing. I'd probably expect us to:

* start with official low-level RPC-style API which is network-agnostic (this I think what `near-json-rpc-client` is)? This thing needs to be robust and complete, but can be ugly and verbose to use
* add official testing DSL which is nice to use, but not necessary maximally complete or robust
* experiment with unofficial or semi-official "nice" clints for the mainnet.

Reply:
That's a very valid point, but the issue is just that with the need for users to interact with mainnet to be able to spoon state or pull any data they can query for into the testing environment, my intuition is that we might not want to be too restrictive on what they can do.

Possible that we might want to disallow sending transactions through mainnet, but given how intentional it has to be to set up a key to do so, not sure it's really that much of a foot gun? @ChaoticTempest maybe you have some input with this

Reply:
The thing was that a good amount of users wanted the ability to be able to migrate their workflow from a testing environment to a live/prod environment so that they don't have to redo it all over again. That way they can easily migrate their flow having properly tested it in sandbox and having some confidence that it won't fail. I can see us disabling things or hiding things behind feature gates if things don't align well with testing like sending `mainnet` transactions though